static/js/disaster_dashboard/map_section.tsx

```diff
@@ -0,0 +1,252 @@
+    point: DisasterEventPoint,
+    event: PointerEvent
+  ): void {
+    const infoCard = d3.select(infoCardRef.current);
+    ReactDOM.render(
+      <InfoCard
+        onClose={() =>
+          d3.select(infoCardRef.current).style("visibility", "hidden")
+        }
+        eventData={point}
+      />,
+      infoCardRef.current
+    );
+    const infoCardHeight = (
+      infoCard.node() as HTMLDivElement
+    ).getBoundingClientRect().height;
+    const infoCardWidth = (
+      infoCard.node() as HTMLDivElement
+    ).getBoundingClientRect().width;
+    const containerWidth = (
+      d3.select(svgContainer.current).node() as HTMLDivElement
+    ).getBoundingClientRect().width;
+    let left = Math.min(event.offsetX, containerWidth - infoCardWidth);
+    if (left < 0) {
+      left = 0;
+      infoCard.style("width", containerWidth + "px");
+    } else {
+      infoCard.style("width", "fit-content");
+    }
+    let top = event.offsetY - infoCardHeight - iNFO_CARD_OFFSET;
+    if (top < 0) {
+      top = event.offsetY + iNFO_CARD_OFFSET;
+    }
+    infoCard
+      .style("left", left + "px")
+      .style("top", top + "px")
+      .style("visibility", "visible");
+  }
+
+  function canClickRegion(placeDcid: string): boolean {
+    const enclosingPlace =
+      props.europeanPlaces.findIndex((country) => country.dcid === placeDcid) >
+      -1
+        ? EUROPE_NAMED_TYPED_PLACE
+        : props.selectedPlaceInfo.place;
+    const parentPlaces = getParentPlaces(
+      props.selectedPlaceInfo.place,
+      enclosingPlace,
+      props.selectedPlaceInfo.parentPlaces
+    );
+    const newPlaceAsNamedTypedPlace = {
+      dcid: placeDcid,
+      name: placeDcid,
+      types: [props.selectedPlaceInfo.placeType],
+    };
+    return !_.isEmpty(
+      getAllChildPlaceTypes(newPlaceAsNamedTypedPlace, parentPlaces).filter(
+        (placeType) =>
+          placeType !== IPCC_PLACE_50_TYPE_DCID &&
+          placeType !== props.selectedPlaceInfo.placeType
+      )
+    );
+  }
+
+  function draw(): void {
+    const width = svgContainer.current.offsetWidth;
+    const height = (width * 2) / 5;
+    const zoomParams = {
+      zoomInButtonId: ZOOM_IN_BUTTON_ID,
+      zoomOutButtonId: ZOOM_OUT_BUTTON_ID,
+    };
+    document.getElementById(MAP_ID).innerHTML = "";
+    const isUsaPlace = isChildPlaceOf(
+      props.selectedPlaceInfo.place.dcid,
+      USA_PLACE_DCID,
+      props.selectedPlaceInfo.parentPlaces
+    );
+    const projection = getProjection(isUsaPlace, "", width, height);
+    drawD3Map(
+      MAP_ID,
+      props.geoJson,
+      height,
+      width,
+      {},
+      "",
+      null,
+      (geoDcid: GeoJsonFeatureProperties) => {
+        const namedPlace = {
+          name: geoDcid.name,
+          dcid: geoDcid.geoDcid,
+        };
+        props.onPlaceUpdated(namedPlace);
+      },
+      () => "",
+      canClickRegion,
+      false,
+      true,
+      projection,
+      "",
+      "",
+      zoomParams
+    );
```

Comment:
nit: since this is so long now, add comments for some of these arguments? especially the ones that are skipped

Reply:
done